pkg/dns/plugin/nsmap/namespace.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nsmap
+
+import (
+	"context"
+	"strings"
+
+	"github.com/coredns/coredns/plugin/rewrite"
+	"github.com/coredns/coredns/request"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+)
+
+type namespaceRewriter struct {
+	Namespaces map[string]string
+}
+
+func (m *namespaceRewriter) updateFromConfigmap(cm *corev1.ConfigMap) {
+	klog.Infof("reloading nsmap config map : +%v", cm.Data)
```

Comment:
Please have this take in a `ctx` so you can get the logger from it. We are not adding new non-contextual `klog` statements if we can avoid it.

Reply:
done.